index.html

```diff
@@ -244,9 +269,71 @@
         </dt>
       </dl>
       <p>
-        The <code>PermissionName</code> enum defines the list of known
-        permission names.
+        Each enumerator-value in the <a>PermissionName</a> enum identifies a
+        <dfn data-export="">permission</dfn>, which consists of the following
+        algorithms and types:
       </p>
+      <dl>
+        <dt>
+          An <dfn data-export="">associated PermissionDescriptor</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionDescriptor</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission storage type</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionStorage</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission result type</dfn>
+        </dt>
+        <dd>
+          A subtype of <a>PermissionStatus</a>.
+        </dd>
+        <dt>
+          An algorithm to <dfn data-export="">get a permission storage
+          identifier</dfn>.
+        </dt>
+        <dd>
+          Takes an <a>environment settings object</a> argument and returns a
+          <a>permission storage identifier</a>.
+        </dd>
+        <dt>
+          A <dfn data-export="">permission query algorithm</dfn>
+        </dt>
+        <dd>
+          Takes an <a>associated PermissionDescriptor</a>, an instance of the
```

Comment:
I think this would be "an instance of the associated permission descriptor type for this permission". The main thing is instance; right now associated PermissionDescriptor/associated permission descriptor type is a type, not an object.